sphinx_gallery/backreferences.py

```diff
@@ -394,12 +426,16 @@ def _write_backreferences(
                     target_dir,
                     gallery_conf["src_dir"],
                     fname,
-                    snippet,
+                    intro,
                     title,
                     is_backref=True,
                 )
             )
             seen_backrefs.add(backref)
+            backrefs_example[backref].append(
+                (str(Path(target_dir, fname)), intro, title)
+            )
+    return backrefs_example
```

Comment:
I think changing to a proper `dict` instead of `defaultdict` is a good idea because it will make it raise KeyError rather than add an entry if you try to access a non-existent key
```suggestion
    return dict(backrefs_example)
```

Reply:
Happy to do this but currently we only ever use `_combine_backreferences` on the returned backreferences - which I think returns a normal dict. I guess if we ever use it for something else in the future it would be useful?

Reply:
Amended just to be safe!